src/Forms/Form.php

```diff
@@ -1552,10 +1552,13 @@ public function forTemplate()
             HTTPCacheControlMiddleware::singleton()->disableCache();
         }
 
-        $return = $this->renderWith($this->getTemplates());
+        $context = $this;
```

Comment:
does assigning `$this` to `$context` do anything? `$this` is an object and so will be assigned by reference to `$context`...

Reply:
Yep, but storing it in a new variable allows that variable to be overwritten when an extension hook is created with an argument passed by reference:

```php
public function onBeforeRender(Form &$form)
{
    $form = new Form();
    return $form;
}
```

Without `$context = $this` that won’t work. I can’t think of a use-case for this at a form level, but I wanted to keep it consistent with form fields

Reply:
Dumb example: https://3v4l.org/EHc1E